### PR TITLE
fix: eliminate TP-first selection bias in few-shot precedents (#264)

### LINKED ISSUE #264: query_feedback_precedents under-fills and biases toward TPs
## Description

`query_feedback_precedents` in `src/pipeline.rs` (L255-278) queries the feedback index for precedents but may under-fill the result set and bias toward TP entries. The current selection logic doesn't ensure a balanced mix of TP and FP precedents for few-shot injection, which can anchor the LLM toward finding more issues rather than exercising appropriate skepticism.

## Impact

LLM review may over-report findings in areas with dense TP feedback history, reducing precision.

Found by quorum self-review.

### PR DESCRIPTION
**Pull request by jsnyder**
## Summary

- **Replaces TP-first selection** with similarity-rank interleaving: candidates are now taken in descending similarity order regardless of verdict, eliminating anchoring bias that pushed the LLM toward over-reporting
- **Adds diversity floor**: guarantees at least 1 TP + 1 FP in the 3-slot window when both exist in the candidate pool, preventing homogeneous few-shot examples
- **Adds relative similarity threshold**: candidates below 60% of the best score are filtered out, preventing weak matches from polluting the prompt
- Single file change (`src/pipeline.rs`), 12 new unit tests covering all branches

## Design

See `docs/plans/2026-05-10-issue-264-few-shot-bias.md` for the full implementation plan. Key design reviewed by Gemini 3.1 Pro before implementation.

Algorithm: `select_few_shot_precedents(candidates)` — takes pre-sorted candidates, applies threshold filter, fills 3 slots in similarity order with a diversity constraint that reserves the last slot for the minority verdict if needed.

## Quorum self-review verdicts

| Finding | Verdict | Reason |
|---------|---------|--------|
| `select_few_shot_precedents` cyclomatic complexity 17 | `fp` (out-of-scope) | Inherent to selection logic; well-tested with 12 unit tests |
| `silent-error-conversion` (lines 505-508) | `fp` (out-of-scope) | Pre-existing HOME fallback, not touched by this branch |

## Test plan

- [x] 12 unit tests covering: similarity-rank ordering, diversity floor (TP-heavy, FP-heavy), homogeneous pools, threshold filtering, boundary conditions, empty/single/two candidates, diversity skip past position 3, minority below threshold
- [x] `cargo test --bin quorum` — all 1746 tests pass
- [x] `cargo clippy` — clean
- [x] `cargo build --release` — succeeds

Closes #264

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Co-Authored-By: Claude Opus 4.6 <noreply@anthropic.com>

## Summary by CodeRabbit

* **Improvements**
  * Enhanced feedback precedent selection with improved diversity enforcement and similarity-based filtering
  * More intelligent selection logic for few-shot examples from ranked candidates

* **Tests**
  * Extended test suite with comprehensive unit tests covering selection ordering, diversity handling, threshold filtering, and edge cases

[Review Change Stack](https://app.coderabbit.ai/change-stack/jsnyder/quorum/pull/283)

### CODE REVIEW

File: src/pipeline.rs [hunk 1]
```diff
@@ -246,8 +246,94 @@ fn truncate_for_review(source: &str, max_lines: usize) -> (String, Option<String
...
+        "select_few_shot_precedents expects only Tp/Fp candidates; filter upstream"
+    );
+
+    let best_score = candidates[0].similarity;
+    let threshold = best_score * 0.6;
+
+    let viable: Vec<_> = candidates
+        .iter()
+        .filter(|s| s.similarity >= threshold)
+        .collect();
```
[the remark this answers is not in the document]
Comment (jsnyder) [reply src/pipeline.rs:288]: Applied in 2687b2d. Added `debug_assert!` for descending similarity sort precondition.
